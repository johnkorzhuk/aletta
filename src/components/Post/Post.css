.title {
  font-size: 3rem;
  color: #2f89de;
}

.content {
  font-size: 1.8rem;
  line-height: 1.7
}

.link {
  color: #2f89de; 
}

.link:focus,
.link:hover {
  color: #30699e;
  transition: color 0.3s ease;
}
