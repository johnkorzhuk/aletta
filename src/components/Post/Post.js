import React, { PropTypes } from 'react'
import { Link } from 'react-router-dom'
const { string, bool } = PropTypes

import styles from './Post.css'

const Post = ({
  title,
  excerpt,
  content,
  slug,
  titleLink
}) => {
  return (
    <div>
      <h2 className={styles.title}>
        {
          titleLink
            ? <Link className={styles.link} to={`/posts/${slug}`}>{title}</Link>
            : title
        }
      </h2>
      <p className={styles.content}>{excerpt}</p>
    </div>
  )
}

Post.propTypes = {
  title: string,
  excerpt: string,
  content: string,
  slug: string,
  titleLink: bool
}

export default Post
