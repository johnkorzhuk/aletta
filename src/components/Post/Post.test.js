/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import Post from './Post'

test('Post component should render as expected with title as props', () => {
  const component = shallow(<Post title='Some title' />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})

test('Post component should render as expected with titleLink as props', () => {
  const component = shallow(<Post titleLink />)
  const tree = toJson(component)
  expect(tree).toMatchSnapshot()
})
