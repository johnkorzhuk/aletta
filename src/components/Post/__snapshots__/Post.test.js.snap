// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Post component should render as expected with title as props 1`] = `
<div>
  <h2
    className="title"
  >
    Some title
  </h2>
  <p
    className="content"
  />
</div>
`;

exports[`Post component should render as expected with titleLink as props 1`] = `
<div>
  <h2
    className="title"
  >
    <Link
      className="link"
      replace={false}
      to="/posts/undefined"
    />
  </h2>
  <p
    className="content"
  />
</div>
`;
